### PR TITLE
chore(DTFS2-7458): align facility identifier and facility

### PR DESCRIPTION
**Pull request by abhi-markan**
## Introduction :pencil2:
Ensure variable names are consistent when referring to facility identifier for ACBS pertinent payload.

## Resolution :heavy_check_mark:
* `facilityId` has been renamed to `facilityIdentifier`.

## Miscellaneous :heavy_plus_sign:
* Minor documentation updates.
* Precisely named functions to accurately describe execution actions.
* Added `await` to `updateDeal` (same fix was merged to `release-2.4.0`)

### CODE REVIEW

File: azure-functions/acbs-function/api.js [hunk 1]
```diff
@@ -195,8 +195,8 @@ const createFacilityCovenantId = (payload) => postMdm('numbers', payload);
 // *** TFS ***
 
 // GET
-const getFacility = (facilityId) => get(`facilities/${facilityId}`);
-const getLoanId = (facilityId) => get(`facilities/${facilityId}/loans`);
+const getFacility = (facilityIdentifier) => get(`facilities/${facilityIdentifier}`);
```
Comment (ttbarnes) [inline azure-functions/acbs-function/api.js:198]: Can we add some documentation here with examples of some facility identifiers? Something like 

```ts
@param {String} facilityIdentifier - Facility ACBS ID - "Bond" = 1, "Cash" = 2
```

File: azure-functions/acbs-function/src/functions/acbs-amend-facility.js [hunk 2]
```diff
@@ -73,12 +73,12 @@ df.app.orchestration('acbs-amend-facility', function* amendFacility(context) {
     const { facility, deal } = amendment;
     const { facilitySnapshot } = facility;
 
-    if (facilityId.includes(DEAL.UKEF_ID.PENDING) || facilityId.includes(DEAL.UKEF_ID.TEST)) {
-      throw new Error(`Invalid facility ID ${facilityId}`);
+    if (facilityIdentifier.includes(DEAL.UKEF_ID.PENDING) || facilityIdentifier.includes(DEAL.UKEF_ID.TEST)) {
+      throw new Error(`Invalid facility ID ${facilityIdentifier}`);
```
Comment (ttbarnes) [inline azure-functions/acbs-function/src/functions/acbs-amend-facility.js:77]:
```suggestion
      throw new Error(`Invalid facility identifier ${facilityIdentifier}`);
```

File: azure-functions/acbs-function/src/functions/acbs-issue-facility.js [hunk 3]
```diff
@@ -85,20 +85,20 @@ df.app.orchestration('acbs-issue-facility', function* issueFacility(context) {
...
           facilityFee = yield context.df.callActivityWithRetry('create-facility-fee', retryOptions, {
-            facilityIdentifier: facilityId,
+            facilityIdentifier,
             acbsFacilityFeeInput,
           });
         }
 
         return {
-          facilityId: facility._id,
+          facilityIdentifier: facility._id,
```
Comment (ttbarnes) [inline azure-functions/acbs-function/src/functions/acbs-issue-facility.js:101]: Finding this a bit confusing without context, why is `facilityIdentifier` here not the same as other instances? Maybe could add a comment?

**Review by ttbarnes — CHANGES_REQUESTED** [review]
Nice work, few minor comments/suggestions

File: azure-functions/acbs-function/api.js (hunk 1 above)
Comment (abhi-markan) [reply azure-functions/acbs-function/api.js]: Have added to all the functions.

File: azure-functions/acbs-function/src/functions/acbs-issue-facility.js (hunk 3 above)
Comment (abhi-markan) [reply azure-functions/acbs-function/src/functions/acbs-issue-facility.js]: Good spot, I missed it out have removed `facility._id`.

**Review by ttbarnes — APPROVED** [review]

**Pull request merged**